plugins/source/test/resources/services/test_sub_table.go

```diff
@@ -53,6 +53,9 @@ func testSubTable(config client.Spec) *schema.Table {
 
 func fetchSubTableData(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
 	cl := meta.(*client.Client)
+	if cl.Spec.FailImmediately {
+		return fmt.Errorf("failing immediately")
```

Comment:
WDYT about having a defined error that we return in all three of these resolvers? e.g. `var ErrFailImmediately = ...; return ErrFailImmediately`

Reply:
Makes sense. Implemented.